pyiron_base/jobs/master/generic.py

```diff
@@ -164,6 +166,14 @@ def child_project(self):
         else:
             return self.project.open(self.job_name + "_hdf5")
 
+    @property
+    def input(self):
+        return self._input
+
+    @input.setter
+    def input(self, new_input):
+        self._input = new_input
```

Comment:
```suggestion
    def input(self, new_input: GenericParameters):
        if isinstance(new_input, GenericParameters):
            self._input = new_input
        else:
            raise TypeError(f"Expected a GenericParameters object but got {new_input.__class__})
```

If we're going to bother making the input variable private, let's exploit that to do some type checking.